### PR TITLE
move image digests resolution to backend

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
introduced `ResolveImageDigests` in the API and moved image digest resolution to the backend
longer terms goal is to use `dockercli.RegistryClient()` to replace `cnab-to-oci` dependency

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

File: cmd/compose/convert.go [hunk 1]
```diff
@@ -141,8 +141,9 @@ func runConvert(ctx context.Context, backend api.Service, opts convertOptions, s
 	}
```
Comment (milas) [inline cmd/compose/convert.go:141]: Delete lines 130-141 now?

**Review by milas — APPROVED** [review]
LGTM - small comment re: duplicated code

File: cmd/compose/convert.go (hunk 1 above)
Comment (ndeloof) [reply cmd/compose/convert.go]: oh yes indeed, my bad

**Comment by ndeloof** [comment]
@thaJeztah thanks, this is exactly what I was looking for! 🥰

**Comment by thaJeztah** [comment]
Make sure to double check in the cli as well as at some point resolution was moved; check for "no-resolve-image" (I think) on the docker cli (I'm just away from my keyboard right now 😬😅)

**Review by milas — APPROVED** [review]
✨

File: pkg/compose/compose.go [hunk 2]
```diff
@@ -93,13 +94,34 @@ func getContainerNameWithoutProject(c moby.Container) string {
...
+	}
+
 	switch options.Format {
 	case "json":
 		return json.MarshalIndent(project, "", "  ")
 	case "yaml":
 		return yaml.Marshal(project)
 	default:
-		return nil, fmt.Errorf("unsupported format %q", options)
+		return nil, fmt.Errorf("unsupported format %q", options.Format)
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply pkg/compose/compose.go:124]: linter suddenly noticed this bug, while it existed before ... weird

**Pull request merged**